src/chains/filecoin/filecoin/src/blockchain.ts

```diff
@@ -214,8 +245,8 @@ export default class Blockchain extends Emittery.Typed<
       state: DealState.Validating, // Not sure if this is right, but we'll start here
       message: "",
       provider: this.miner,
-      pieceCid: null,
-      size: size,
+      pieceCid: proposal.data.pieceCid,
+      size: proposal.data.pieceSize || size,
```

Comment:
It looks like the `let size = await this.getIPFSObjectSize(proposal.data.root["/"].value);` line isn't necessary if `proposal.data.pieceSize` is set. Can you adjust so we don't fetch the `size` if we don't need it?

Reply:
`proposal.data.pieceSize` is an optional parameter and I have seen examples that even have provided it to be `0` instead of undefined to indicate "the miner should calculate it". I can change the type to have `pieceSize?` (which may be a rabbit hole with the current way these types work), but we'd still need to fetch the size.